python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py

```diff
@@ -366,9 +366,12 @@ def ensure_schema_exists(context: OutputContext, table_slice: TableSlice, connec
     def delete_table_slice(context: OutputContext, table_slice: TableSlice, connection) -> None:
         try:
             connection.execute(_get_cleanup_statement(table_slice))
-        except ProgrammingError:
-            # table doesn't exist yet, so ignore the error
-            pass
+        except ProgrammingError as e:
+            if "does not exist" in e._message():  # type: ignore # noqa: SLF001
```

Comment:
pyright seems to think `e._message()` doesnt exist, but I can go up the inheritance stack of `ProgrammingError` and see it. Not sure if I should be doing something else rather than ignoring the pyright complaint though